client/src/quic_client.rs

```diff
@@ -90,10 +90,11 @@ impl TpuConnection for QuicTpuConnection {
         Ok(())
     }
 
-    fn send_wire_transaction_async(&self, wire_transaction: Vec<u8>) -> TransportResult<()> {
+    fn send_wire_transaction_async(&self, wire_transaction: &[u8]) -> TransportResult<()> {
```

Comment:
Won't this mean that if the caller had a vector, they still have to pass a slice, which gets copied to a new vector inside this function?

In the following example, the `send_transaction()` had a vector (`wire_transaction`). So it could have transferred the ownership to `send_wire_transaction_async()`. But since now `send_wire_transaction_async` expects a slice, it will create a copy of the original vector inside `send_wire_transaction_async` function?

```
    pub fn send_transaction(
        &self,
        transaction: &Transaction,
    ) -> std::result::Result<(), std::io::Error> {
        let wire_transaction = serialize(transaction).expect("serialization should succeed");
        send_wire_transaction_async(&wire_transaction, ...)
    }
```

Reply:
Yes, that's true, and I originally felt that we don't care, because all the callers are doing this anyway. The SendTransactionService, for example, stores the wire_transaction in its transactions map, so it always has to be reallocated to send.

But I concede.